### ISSUE TITLE
Can you expand on what this does?

### ISSUE THREAD

**Issue by MatthiasKauer** [vulpino/dit#2]
Hi,
I saw your project on onethingwell and now I'm wondering whether I correctly understand what it's supposed to do.

When you say "dotfile", you mean files like .bashrc .bash_aliases, .gdbinit in my home folder, right?
So I would put these files in a git repository somewhere, say ~/my_repos/my_dotfiles/ and then symlink to that. Instead of doing that manually, your project adjusts the links correctly when I pull new files I edited elsewhere?

Another question regarding the setup: where does `dit init` need to be executed? I suppose in the new repo location because the home folder is easier to find.
So basically, it would then be

```
mkdir ~/my_repos/my_dotfiles/ 
cd ~/my_repos/my_dotfiles/
dit init 
```

Can you comment on that?
Kind regards,
Matthias

**Comment by fluffgfx** [comment]
1. That is correct - dit uses the built in git hooks system, so it symlinks any new files to your home directory every time a `git commit` or `git pull` is made.
2. Yes - you execute `dit init` in the folder containing your dotfiles. I'll update the readme to reflect this.

Any further questions I would be happy to answer.

**Comment by MatthiasKauer** [comment]
Thanks for the explanation!

Before I try this, I'm also wondering what will happen with existing files. Is it possible that the script will just overwrite them when symlinking?

Can you also say a few words on how this relates to existing tools like [homesick](https://github.com/technicalpickles/homesick)? Was the hook-based symlinking your main purpose?

**Comment by fluffgfx** [comment]
The script doesn't know how to handle existing files and will probably fail to symlink any existing files. I'll add a prompt in case a file exists on whether to overwrite the existing file/symlink tomorrow, for version 0.3.

As to comparing to homesick, homesick is much more feature rich than dit - dit is meant to be simple and painless, whereas homesick is incredibly complex (having a full suite of commands, support for multiple "castles," .homesickrc, etc). If you need multiple dotfiles repositories or features like .homesickrc and .homesick_subdir, use homesick. If you just need something easy and quick, use dit.

Git hooks are just an easy way of ensuring new files are symlinked into the home directory upon commit or pull. They're an incredibly convenient and terribly underutilized system, and none of the tools I found for managing dotfiles used git hooks.

As an aside, I would wait for version 0.3 before using this, since I need to add the check for existing symlinks mentioned above (It'll just spit out a bunch of "Failed to symlink ~/.dotfile" messages if I recall correctly) and a `dit clean` command mentioned in #3 to remove existing dead symlinks. I'll be working on this over the week and should have a functioning version by Friday - any further suggestions would be welcome.

**Comment by fluffgfx** [comment]
As there's been no activity on this issue for a week I'm closing it.
